source/_posts/2015-03-26-creating-a-dynamic-single-page-app-for-our-genome-team-using-react.markdown

```diff
@@ -0,0 +1,96 @@
+---
+layout: post
+title: "Creating a dynamic single page app for our Genome Team using React"
+date: 2015-03-26 10:42
+comments: true
+categories: [React]
+author: Sarah Weir
+github-url: https://www.github.com/sweir27
+twitter-url: http://twitter.com/sweir27
+---
+We recently picked up a Rails application that was a few bugs away from completion. This application allows our [Genome Team](https://www.artsy.net/about/the-art-genome-project) to genome (classify artworks based on visual and art historical characteristics) multiple artworks at once. These characteristics, or "genes" can be added, removed and changed for any of the artworks on the panel. There are many ways to find and add genes to an artwork in this app. For example, a genomer may pull up an existing artist genome to use as a 'template' from which they can batch apply certain genes to multiple artworks. With these and other dynamic views required, our consumers desired an interaction-heavy single page app.
+
+![screenshot](http://cl.ly/image/310O3U3J3T2c)
+
+The backend was organized, modular, and interfaced seamlessly with the Artsy API. However, the single page app needs of our customer were met with a multitude of javascript-heavy files declaring global event listeners, needless to say, these were clunky and difficult to maintain. When we were tasked with resurrecting this app by fixing the critical bugs preventing it from being adopted, we found that simple requests took much more time than they should, and often resulted in a shamefully hacky fix. We intended to contribute as little work as possible up front in order to get the app out the door as soon as possible, and then to refactor retroactively. After a month of frustration with no end to bugs in sight, we decided to stop trying to patch our leaky roof and build a new one.
+
+### Choosing a suitable framework 
+
+We spent a day researching different front end frameworks to see what would best fit our needs. Our requirements were:
+- A robust view layer that could work on top of our already-solid Rails backend
+- Performant enough for an interaction-heavy single-page app, with hundreds of editable fields autosaving on change
+- We preferred light, streamlined frameworks, and valued freedom over unnecessary structure
```

Comment:
We preferred read strange here, since it's supposed to be a requirement. Maybe "A streamlined framework that favors freedom over unnecessary structure.